app/controllers/passwords_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PasswordsController < ApplicationController
+
+  def index; end
+
+  def create 
+    @user = User.find_by(email: params[:email])
+      if @user.present?
+        #send email
+        PasswordMailer.with(user: @user).reset.deliver_now
+      end
+    redirect_to password_path, notice: "If an account with that email was found, we have sent a link to reset your password"
+  end
+    
+  def edit
+    @user = User.find_signed!(params[:token], purpose: "password_reset")
```

Comment:
use filter to find user. Example - before_action

Reply:
I think we can not get user in PasswordController using before action  because we will not be login if we forget password

Reply:
For both Update and edit action you are using same command optimise your code by using filters!

Reply:
I have pushed it again with this change